File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem5.cpp
#include <iostream>
using namespace std;

struct stInfo {
	int Age;
	bool HasDrivingLicense;
	bool HasRecommendation;
};
stInfo ReadInfo() {
	stInfo Info;
	cout << "Please enter your Age : ";
	cin >> Info.Age;

	cout << "Do you have a Driving license : ";
	cin >> Info.HasDrivingLicense;

	cout << "Do you have a Recommendation : ";
	cin >> Info.HasRecommendation;

	return Info;
}

bool IsAccepted(stInfo Info) {
	if (Info.HasRecommendation)
		return true;
	else
		return (Info.Age > 21 && Info.HasDrivingLicense);
}
void PrintResult(stInfo Info) {
	if (IsAccepted(Info))
		cout << "\n Hired";
	else
		cout << "\n Rejected";
}
int main() {
	PrintResult(ReadInfo());
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem5.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        struct stInfo
        {
            public int Age;
            public bool HasDrivingLicense;
            public bool HasRecommendation;
        }
        static stInfo ReadInfo()
        {
            stInfo Info = new stInfo();

            Console.Write("\nPlease enter your age : ");
            Info.Age = Int32.Parse(Console.ReadLine());

            // enter (true) or (false)
            Console.Write("\nDo you have a driving license : ");
            Info.HasDrivingLicense = Convert.ToBoolean(Console.ReadLine());

            // enter (true) or (false)
            Console.Write("\nDo you have a Recommendation  : ");
            Info.HasRecommendation = Convert.ToBoolean(Console.ReadLine());

            return Info;
        }
        static bool IsAccepted(stInfo Info)
        {
            if(Info.HasRecommendation)
                return true;
            else
                return Info.HasDrivingLicense && Info.Age > 21;
            
        }
        static void PrintResult(stInfo Info)
        {
            if (IsAccepted(Info))
            {
                Console.WriteLine("\nHired");
            }
            else
            {
                Console.WriteLine("\nRejected");
            }
        }
        static void Main(string[] args)
        {
            PrintResult(ReadInfo());
        }
    }
}